server/consumer.go

```diff
@@ -4262,26 +4249,6 @@ func deliveryFormsCycle(cfg *StreamConfig, deliverySubject string) bool {
 	return false
 }
 
-// Check that the filtered subject is valid given a set of stream subjects.
-func validFilteredSubject(filteredSubject string, subjects []string) bool {
-	if !IsValidSubject(filteredSubject) {
```

Comment:
We still want this check.

Reply:
What is the reason we still need this check?

The two reasons I removed it being:
- with subject transform in the stream it's no longer a valid test: a stream could be listening on subject "foo.>" and then have an input transform that modifies the subject to (for example) "bar.>", so all the messages in the stream will be under "bar.>", so creating a consumer (or sourcing from that stream) with a filter "foo.>" (or subset thereof) would pass that check and yet the consumer would not be able to get any message from the stream (while using a filter "bar.>" would fail the check and yet is an appropriate filter to get the messages in the stream).
- This check also prevents sourcing from a stream that is itself sourced (or mirrored) because a stream that has sourcing or mirroring another stream may not have a subject filter at all, or if it does it could be on a completely different subject than what is being sourced. This is what @tbeets is talking about in his comment above.

Reply:
That first check checks that its a valud subject and not something like `foo..bar` or `.foo ` or `foo.>.bar`

Reply:
I just want us to execute line 4267.

Reply:
Oh I see what you mean, yes good catch. Done.